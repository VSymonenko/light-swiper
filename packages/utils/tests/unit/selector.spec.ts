import { detectSelector, getElement } from '@/index';

describe('should be detect type of selector', () => {
  test('type selector should be is id', () => {
    expect(detectSelector('#test')).toEqual('id');
  });

  test('type selector should be is class', () => {
    expect(detectSelector('.test')).toEqual('class');
  });
});

describe('should be return e DOM element', () => {
  test('should return HTMLElement', () => {
    const id = 'test';
    const div = document.createElement('div');
    div.id = id;
    document.body.appendChild(div);
    const element = getElement('id', `#${id}`);

    expect(element).toBeInstanceOf(HTMLElement);

    expect(element).toEqual(div);

    expect(element).toMatchSnapshot();
  });

  test('should be return Element', () => {
    const swiperClass = 'test';
    const div = document.createElement('div');
    div.className = swiperClass;
    document.body.appendChild(div);
    const element = getElement('class', `.${swiperClass}`);

    expect(element).toBeInstanceOf(HTMLDivElement);

    expect(element).toEqual(div);

    expect(element).toMatchSnapshot();
  });

  test('should be a null', () => {
    const element = getElement('class', '');

    expect(element).toBeNull();
  });
});
