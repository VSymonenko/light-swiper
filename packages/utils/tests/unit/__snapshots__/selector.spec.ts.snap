// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be return e DOM element should be return Element 1`] = `
<div
  class="test"
/>
`;

exports[`should be return e DOM element should return HTMLElement 1`] = `
<div
  id="test"
/>
`;
